### PR TITLE
add sj (json seek history (fixes #7663)

### LINKED ISSUE #7663: sj shuold return an array of objects
showing naming of each offset too

### PR DESCRIPTION
**Pull request by lowlyw**

### CODE REVIEW

File: libr/core/cmd_seek.c [hunk 1]
```diff
@@ -347,9 +347,58 @@ static int cmd_seek(void *data, const char *input) {
...
+					r_list_append (names, strdup (name));
+					++lsz;
+					free (name);
+				}
+				r_list_free (list);
+			}
+			r_cons_printf ("[");
+			for (i = 0; i < lsz; ++i) {
+				uintptr_t addr = (uintptr_t)r_list_get_n (addrs, i);
+				char *name = r_list_get_n (names, i);
```
Comment (radare) [inline libr/core/cmd_seek.c]: const

File: libr/core/cmd_seek.c [hunk 2]
```diff
@@ -347,9 +347,58 @@ static int cmd_seek(void *data, const char *input) {
...
+					r_list_append (addrs, (void *)undo->off);
+					r_list_append (names, strdup (name));
+					++lsz;
+					free (name);
+				}
+				r_list_free (list);
+			}
+			r_cons_printf ("[");
+			for (i = 0; i < lsz; ++i) {
+				uintptr_t addr = (uintptr_t)r_list_get_n (addrs, i);
```
Comment (radare) [inline libr/core/cmd_seek.c]: uintptr_t wat, use ut64, but dont depend on pointer sizes

File: libr/core/cmd_seek.c [hunk 3]
```diff
@@ -347,9 +347,58 @@ static int cmd_seek(void *data, const char *input) {
...
+						} else {
+							name = strdup (f->name);
+						}
+					}
+					if (!name) {
+						name = strdup ("");
+					}
+					r_list_append (addrs, (void *)undo->off);
+					r_list_append (names, strdup (name));
+					++lsz;
```
Comment (radare) [inline libr/core/cmd_seek.c]: dont preincrement unnesesarily

File: libr/core/cmd_seek.c [hunk 4]
```diff
@@ -347,9 +347,58 @@ static int cmd_seek(void *data, const char *input) {
...
+					core->flags->space_strict = false;
+					if (f) {
+						if (f->offset != undo->off) {
+							name = r_str_newf ("%s + %d\n", f->name,
+									(int)(undo->off- f->offset));
+						} else {
+							name = strdup (f->name);
+						}
+					}
+					if (!name) {
```
Comment (radare) [inline libr/core/cmd_seek.c]: i think this goes into the else block in there

File: libr/core/cmd_seek.c (hunk 1 above)
Comment (lowlyw) [inline libr/core/cmd_seek.c]: done

File: libr/core/cmd_seek.c (hunk 2 above)
Comment (lowlyw) [inline libr/core/cmd_seek.c]: done

File: libr/core/cmd_seek.c (hunk 3 above)
Comment (lowlyw) [inline libr/core/cmd_seek.c]: huh? whatever

File: libr/core/cmd_seek.c (hunk 4 above)
Comment (lowlyw) [inline libr/core/cmd_seek.c]: nope, it doesn't, this is the case where it's no flags here, so we make an empty.

**Comment by lowlyw** [comment]
merged

**Pull request closed without merging**